internal/wireguard/wg.go

```diff
@@ -24,6 +24,23 @@ var (
 	cleanDNSPattern = regexp.MustCompile(`[^a-zA-Z0-9\\-]`)
 )
 
+func gernateConnectionnName(apiClient *api.Client, ctx context.Context) (string, error) {
```

Comment:
This should be named correctly, perhaps with a better name like `generatePeerName`

Reply:
Also, the context should be the first argument passed in.